httpserver/server.go

```diff
@@ -30,9 +31,17 @@ type Server struct {
 	logger             *logging.Logger
 	metricsRegistry    *metricsd.Registry
 	prometheusExporter *metricsd.PrometheusExporter
-	tokenCache         *TokenCache     // Cache of validated tokens and their session caches
-	oauth2Provider     *OAuth2Provider // OAuth2 provider for MCP endpoints
-	oauth2Config       *oauth2.Config  // OAuth2 client config for SSO
+	tokenCache         *TokenCache          // Cache of validated tokens and their session caches
+	oauth2Provider     *OAuth2Provider      // OAuth2 provider for MCP endpoints
+	oauth2Config       *oauth2.Config       // OAuth2 client config for SSO
+	validatedTokens    map[string]TokenInfo // Cache of validated tokens for rate limiting
```

Comment:
If possible, merge the `tokenCache` and `validatedTokens` together as they have overlapping responsibilities.